meshtastic/interfaces/ble/interface.py

```
@@ -1268,6 +1282,164 @@ def find_device(self, address: str | None) -> BLEDevice:
+
+    def _format_bluetoothctl_address(self, address: str) -> str:
+        """Canonicalize a BLE address for bluetoothctl commands."""
+        sanitized = sanitize_address(address)
+        if sanitized is None or len(sanitized) != 12:
+            raise self.BLEError(
+                f"Cannot trust device: {address!r} did not resolve to a BLE address."
+            )
+        octets = [sanitized[index : index + 2].upper() for index in range(0, 12, 2)]
+        return ":".join(octets)
+
+    def pair(self, address: str | None = None, **kwargs: Any) -> Any:
```

Ruff (0.15.2)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `**kwargs`

(ANN401)